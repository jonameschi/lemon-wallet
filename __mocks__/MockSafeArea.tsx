import React, { FC, PropsWithChildren } from 'react';

import { Metrics, SafeAreaProvider } from 'react-native-safe-area-context';

const MOCK_INITIAL_METRICS: Metrics = {
  frame: {
    width: 320,
    height: 640,
    x: 0,
    y: 0,
  },
  insets: {
    left: 0,
    right: 0,
    bottom: 0,
    top: 0,
  },
};

type MockSafeAreaContextProps = {
  initialWindowMetrics?: Metrics | null;
};

const MockSafeAreaContext: FC<PropsWithChildren<MockSafeAreaContextProps>> = ({
  children,
  initialWindowMetrics,
}) => (
  <SafeAreaProvider
    initialMetrics={initialWindowMetrics || MOCK_INITIAL_METRICS}>
    {children}
  </SafeAreaProvider>
);

export { MockSafeAreaContext };
export type { MockSafeAreaContextProps };
