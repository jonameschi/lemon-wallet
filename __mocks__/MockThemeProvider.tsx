import { FC, PropsWithChildren } from 'react';
import { ThemeProvider } from 'styled-components/native';
import { theme } from 'theme/theme';

export const MockThemeProvider: FC<PropsWithChildren> = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);
