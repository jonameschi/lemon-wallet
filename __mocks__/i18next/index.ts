const useMock = (key: string): string => key;

export default {
  t: useMock,
};
