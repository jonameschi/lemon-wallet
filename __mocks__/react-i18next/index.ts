/* eslint-disable import/no-import-module-exports */
import { I18nextProvider, initReactI18next } from 'react-i18next';

const useMock = {
  t: (key: string, options: object | undefined): string =>
    options ? `${key} - ${JSON.stringify(options)}` : key,
};

module.exports = {
  useTranslation: () => useMock,
  I18nextProvider,
  initReactI18next,
};
