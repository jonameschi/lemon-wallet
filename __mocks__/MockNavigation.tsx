import React, { FunctionComponent, PropsWithChildren } from 'react';

import { NavigationContext } from '@react-navigation/native';
import { MockSafeAreaContext, MockSafeAreaContextProps } from './MockSafeArea';
import { MockThemeProvider } from './MockThemeProvider';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const mockNavigationProps: any = {
  navigate: jest.fn(),
  reset: jest.fn(),
  goBack: jest.fn(),
  isFocused: () => true,
  addListener: jest.fn(() => () => ({ remove: jest.fn() })),
};

type MockNavigationProps = PropsWithChildren<MockSafeAreaContextProps>;

const MockNavigation: FunctionComponent<MockNavigationProps> = ({
  children,
  initialWindowMetrics,
}) => (
  <MockSafeAreaContext initialWindowMetrics={initialWindowMetrics}>
    <NavigationContext.Provider value={mockNavigationProps}>
      <MockThemeProvider>{children}</MockThemeProvider>
    </NavigationContext.Provider>
  </MockSafeAreaContext>
);

export { MockNavigation, mockNavigationProps };
