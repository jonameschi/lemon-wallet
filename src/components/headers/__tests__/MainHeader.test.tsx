import { MainHeader } from '@components';
import { render } from '@testing-library/react-native';
import { MockNavigation } from '../../../../__mocks__/MockNavigation';

const renderTree = () =>
  render(
    <MockNavigation>
      <MainHeader />
    </MockNavigation>,
  );

describe('MainHeader component', () => {
  it('should render a snapshot - default', () => {
    const { toJSON } = renderTree();
    expect(toJSON()).toMatchSnapshot();
  });
});
