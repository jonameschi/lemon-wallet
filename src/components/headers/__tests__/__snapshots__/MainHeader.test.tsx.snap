// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainHeader component should render a snapshot - default 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    paddingTop={10}
    style={
      {
        "backgroundColor": "#2647e4",
        "borderBottomLeftRadius": 30,
        "borderBottomRightRadius": 30,
        "height": 90,
        "paddingBottom": 10,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 10,
        "shadowColor": "rgba(23, 108, 232, 0.175)",
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 4,
      }
    }
  >
    <View
      alignItems="center"
      gap={10}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 10,
          "justifyContent": "flex-start",
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 14,
            "fontWeight": "600",
          }
        }
      >
        welcome - {}
      </Text>
    </View>
    <View
      justifyContent="flex-end"
      style={
        {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "gap": 0,
          "justifyContent": "flex-end",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="logout-btn"
      >
        <View
          pressed={false}
          style={
            {
              "backgroundColor": "#fff",
              "borderBottomLeftRadius": 36,
              "borderBottomRightRadius": 36,
              "borderTopLeftRadius": 36,
              "borderTopRightRadius": 36,
              "opacity": 1,
              "paddingBottom": 8,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 8,
              "width": 150,
            }
          }
        >
          <Text
            style={
              {
                "color": "#212529",
                "fontSize": 16,
                "fontWeight": "600",
                "textAlign": "center",
              }
            }
          >
            logout
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
